server/app/controllers/api/CiviFormApiController.java

```diff
@@ -0,0 +1,31 @@
+package controllers.api;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.AccountNonexistentException;
+import auth.ApiKeyGrants;
+import auth.ProfileUtils;
+import auth.UnauthorizedApiRequestException;
+import controllers.CiviFormController;
+import play.mvc.Http;
+
+/** Base class for controllers that handle API requests */
```

Comment:
This has no abstract methods so it's not really a base class, it's just a class that like all others can be instantiated and expanded.

Do you actually want an interface to reference implementations generically by; or something uninstantiable by making the constructors publicly inaccessible.

Given what I see so far I think you should make the default ctor and the ctor you have protected.